jsonpatcher/jsonpatcher.go

```diff
@@ -50,7 +51,7 @@ func init() {
 
 // New returns a JSON-Patcher EventCodec
 func New() core.EventCodec {
-	return &jsonPatcher{}
+	return &jsonPatcher{jsonMode: true}
```

Comment:
Now the `EventCodec` jsonpatcher should be aware if is working in jsonmode or not.
Basically it tells him to know if `v instance{}` things that receives are Go Types that should marshal, or are string jsons which already can use to do its work.